### PR TITLE
Remove embulk-parser-jsonpath dependency

### PR DESCRIPTION
**Pull request by minidragon88**

### CODE REVIEW

**Comment by huylenq** [comment]
There is little value in testing against our own mock, I'll try to make it run with the real guess implementation.

**Comment by huylenq** [comment]
@tvhung83 could you help taking a look?
Context: https://github.com/treasure-data/embulk-input-jira/pull/53#discussion_r256727766
Mainly removing `embulk-parser-jsonpath` and updating tests.

**Comment by huylenq** [comment]
@minidragon88 since you created the PR and also commited on the branch, it doesn't allow me to add you as a reviewer, but please take a look.

File: src/main/java/org/embulk/input/jira/JiraInputPlugin.java [hunk 1]
```diff
@@ -164,7 +164,7 @@ public ConfigDiff guess(ConfigSource config)
         Set<String> uniqueAttributes = getUniqueAttributes(issues);
         JsonArray samples = createSamples(issues, uniqueAttributes);
         Buffer sample = Buffer.copyOf(samples.toString().getBytes());
-        JsonNode columns = guessExecutor.guessParserConfig(sample, Exec.newConfigSource(), guessConfig).getObjectNode().get("parser").get("columns");
+        JsonNode columns = guessExecutor.guessParserConfig(sample, Exec.newConfigSource(), guessConfig).getObjectNode().get("columns");
```
Comment (tvhung83) [inline src/main/java/org/embulk/input/jira/JiraInputPlugin.java:167]: Not related to this PR, but please make it a habit by not introducing a variable when you don't really need it. For example, `#createGuessConfig` and  `#getGuessExecutor` could be expensive, what a waste if `#validateTaskConfig` fails.

File: src/main/java/org/embulk/input/jira/JiraInputPlugin.java [hunk 2]
```diff
@@ -173,7 +173,7 @@ public ConfigDiff guess(ConfigSource config)
     private ConfigSource createGuessConfig()
     {
         ConfigSource configSource = Exec.newConfigSource();
-        configSource.set("guess_plugins", new ObjectMapper().createArrayNode().add("jsonpath"));
+        configSource.set("guess_plugins", new ObjectMapper().createArrayNode().add("jira"));
```
Comment (tvhung83) [inline src/main/java/org/embulk/input/jira/JiraInputPlugin.java:176]:
```suggestion
        return Exec.newConfigSource()
        .set("guess_plugins", ImmutableList.of("jira"))
        .set("guess_sample_buffer_bytes", GUESS_BUFFER_SIZE);
```
It's easier to read with chaining, besides, you don't need `ArrayNode`, `List` works just fine.

**Review by tvhung83 — CHANGES_REQUESTED** [review]
Left 2 comments. The rest looks good to me.

File: src/main/java/org/embulk/input/jira/JiraInputPlugin.java (hunk 2 above)
Comment (huylenq) [reply src/main/java/org/embulk/input/jira/JiraInputPlugin.java]: I think these should be better addressed on #53, the scope of this PR is about removing `embulk-parser-jsonpath` along with adapting the unit tests.
@tvhung83 how about I clone your comments to #53, proceed to merge this PR (if there aren't any further issue belong to the scope above), then we pick up from there?

**Comment by huylenq** [comment]
I think these: https://github.com/treasure-data/embulk-input-jira/pull/54#discussion_r257113546, https://github.com/treasure-data/embulk-input-jira/pull/54#discussion_r257113913 should be better addressed on #53, the scope of this PR is about removing `embulk-parser-jsonpath` along with adapting the unit tests.
@tvhung83 how about I clone your comments to #53, proceed to merge this PR (if there aren't any further issue belongs to the scope above), then we pick up from there?

File: src/main/java/org/embulk/input/jira/JiraInputPlugin.java (hunk 2 above)
Comment (tvhung83) [reply src/main/java/org/embulk/input/jira/JiraInputPlugin.java]: Ok, please go ahead.

**Review by tvhung83 — APPROVED** [review]

File: src/main/java/org/embulk/input/jira/JiraInputPlugin.java (hunk 1 above)
Comment (huylenq) [reply src/main/java/org/embulk/input/jira/JiraInputPlugin.java:167]: Agree, I also think remove some of the temp variables and inline them would make things more readable.

**Comment by tvhung83** [comment]
Approve to make irrelevant (to this PR) changes in #53

File: src/main/java/org/embulk/input/jira/JiraInputPlugin.java (hunk 1 above)
Comment (huylenq) [reply src/main/java/org/embulk/input/jira/JiraInputPlugin.java:167]: To be addressed on https://github.com/treasure-data/embulk-input-jira/pull/53/files#r257135900

File: src/main/java/org/embulk/input/jira/JiraInputPlugin.java (hunk 2 above)
Comment (huylenq) [reply src/main/java/org/embulk/input/jira/JiraInputPlugin.java:176]: To be addressed on https://github.com/treasure-data/embulk-input-jira/pull/53/files#r257136299

**Pull request merged**